apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -458,5 +484,91 @@ describe('Question Sharing', function () {
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await syncFromDisk.syncOrCreateDiskToSql(sharingCourse.path, logger);
+      const question_id = await sqldb.queryOptionalRow(
+        sql.get_question_id,
+        {
+          course_id: sharingCourse.id,
+          qid: SHARING_QUESTION_QID,
+        },
+        IdSchema,
+      );
+      assert(
+        question_id !== null,
+        'Sync of consuming course should not allow renaming a shared question.',
+      );
+      await fs.rename(questionTempPath, questionPath);
+    });
+
+    step('Sync through the sync page to extablish commit hash.', async () => {
+      await sqldb.queryAsync(sql.update_course_repository, {
+        course_path: sharingCourseLiveDir,
+        course_repository: sharingCourseOriginDir,
+      });
+
+      const syncUrl = `${baseUrl}/course/${sharingCourse.id}/course_admin/syncs`;
+      const response = await fetchCheerio(syncUrl);
+
+      const token = response.$('#test_csrf_token').text();
+      await fetch(syncUrl, {
+        method: 'POST',
+        body: new URLSearchParams({
+          __action: 'pull',
+          __csrf_token: token,
+        }),
+      });
+      const result = await sqldb.queryOneRowAsync(sql.select_last_job_sequence, []);
+      const job_sequence_id = result.rows[0].id;
+
+      await helperServer.waitForJobSequenceStatus(job_sequence_id, 'Success');
+    });
+
+    step('Rename shared question in origin, ensure live does not sync it', async () => {
+      const questionPath = path.join(sharingCourseOriginDir, 'questions', SHARING_QUESTION_QID);
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await execa('git', ['add', '-A'], gitOptions);
+      await execa('git', ['commit', '-m', 'rename shared question'], gitOptions);
+
+      sharingCourse = await selectCourseById(sharingCourse.id);
+
+      const syncUrl = `${baseUrl}/course/${sharingCourse.id}/course_admin/syncs`;
+      const response = await fetchCheerio(syncUrl);
+      const token = response.$('#test_csrf_token').text();
+      await fetch(syncUrl, {
+        method: 'POST',
+        body: new URLSearchParams({
+          __action: 'pull',
+          __csrf_token: token,
+        }),
+      });
+      const result = await sqldb.queryOneRowAsync(sql.select_last_job_sequence, []);
+      const job_sequence_id = result.rows[0].id;
+
+      await helperServer.waitForJobSequenceStatus(job_sequence_id, 'Error');
+
+      assert(
+        sharingCourse.commit_hash === (await selectCourseById(sharingCourse.id)).commit_hash,
+        'Commit hash of sharing course should not change when attempting to sync breaking change.',
+      );
+
+      const sharedQuestionExists = await fs
+        .stat(path.join(sharingCourseLiveDir, 'questions', SHARING_QUESTION_QID))
+        .then(() => true)
+        .catch(() => false);
```

Comment:
Why all this instead of `fs.exists(...)`?

Reply:
it appears to be marked deprecated: https://nodejs.org/api/fs.html#fsexistspath-callback 
but I can still use it if you want.

Reply:
Sorry, I should have recommended `pathExists` specifically (you have imported `fs-extra`, which has a very-not-deprecated function by that name: https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md)

Reply:
Thanks. Gotta love the confusing history of fs/fs-extra

Fixed.